frappe/email/doctype/notification/notification.py

```diff
@@ -624,10 +626,25 @@ def get_receiver_list(self, doc, context, field_on_user="mobile_no", recipient_e
 
 		return list(set(receiver_list))
 
-	def get_attachment(self, doc):
-		"""check print settings are attach the pdf"""
-		if not self.attach_print:
-			return None
+	def get_attachments(self, doc) -> list[dict]:
+		"""Check Attachment Settings and return attachments accordingly"""
+		attachments = []
+		if self.attach_print and (print_info := self.get_print(doc)):
+			attachments.append(print_info)
```

Comment:
```suggestion
		if self.attach_print:
			attachments.append(self.get_print(doc))
```

This is not really needed as part of the if-statement, because `get_print` will never return a falsy value.